optas/models.py

```diff
@@ -198,24 +287,157 @@ def link_names(self):
 
     @property
     def actuated_joint_names(self):
-        return [jnt.name for jnt in self._urdf.joints if jnt.type != 'fixed']
+        return [jnt.name for jnt in self._urdf.joints if jnt.type != "fixed"]
+
+    @property
+    def parameter_joint_names(self):
+        return [joint for joint in self.actuated_joint_names if joint in self.param_joints]
+    
+    @property
+    def parameter_joint_indexes(self):
+        return [self._get_actuated_joint_index(joint) for joint in self.parameter_joint_names]
+
+    @property
+    def optimized_joint_names(self):
+        return [joint for joint in self.actuated_joint_names if joint not in self.parameter_joint_names]
+
+    @property
+    def optimized_joint_indexes(self):
+        return [self._get_actuated_joint_index(joint) for joint in self.optimized_joint_names]
+
+    @property
+    def param_joint_names(self):
+        return [joint for joint in self.actuated_joint_names if joint in self.param_joints]
+    
+    @property
+    def param_joint_indexes(self):
+        return [self._get_actuated_joint_index(joint) for joint in self.param_joint_names]
+
+    @property
+    def opt_joint_names(self):
+        return [joint for joint in self.actuated_joint_names if joint not in self.param_joint_names]
+
+    @property
+    def opt_joint_indexes(self):
+        return [self._get_actuated_joint_index(joint) for joint in self.opt_joint_names]
+
+    @property
+    def parameter_joint_names(self):
+        return [joint for joint in self.actuated_joint_names if joint in self.param_joints]
+    
+    @property
+    def parameter_joint_indexes(self):
+        return [self._get_actuated_joint_index(joint) for joint in self.parameter_joint_names]
+
+    @property
+    def optimized_joint_names(self):
+        return [joint for joint in self.actuated_joint_names if joint not in self.parameter_joint_names]
+
+    @property
+    def optimized_joint_indexes(self):
+        return [self._get_actuated_joint_index(joint) for joint in self.optimized_joint_names]
+
+    @property
+    def parameter_joint_names(self):
+        return [joint for joint in self.actuated_joint_names if joint in self.param_joints]
+    
+    @property
+    def parameter_joint_indexes(self):
+        return [self._get_actuated_joint_index(joint) for joint in self.parameter_joint_names]
+
+    @property
+    def optimized_joint_names(self):
```

Comment:
This is by design. I did it this way so that if you specify a pram_joint that is not contained in the joints of the robot.
It neither gives an error nor does anything wrong.
It just filters out all the requested parametric joints that are not in the list of actuated joints.